lib/workspace.js

```diff
@@ -3,8 +3,25 @@ const request = require('request');
 
 const config = require('./config.js');
 const logger = require('./logger');
+const socketServer = require('./socket-server');
+
+const sqldb = require('@prairielearn/prairielib/sql-db');
+
+module.exports.init = function(callback) {
+    socketServer.io.on('connection', this.connection);
+    callback(null);
+};
 
 module.exports = {
+    updateState(workspace_id, state, callback) {
+        // TODO: add locking
+        sqldb.call('workspaces_state_update', [workspace_id, state], function(err, _result) {
+            if (ERR(err, callback)) return;
+            logger.info(`[workspace.js] set workspaces.state to '${state}'`);
+            socketServer.io.of('/workspace').to(workspace_id).emit('change:state', {workspace_id, state});
```

Comment:
Here we are using socket namespaces (the `of('/workspace')` part). But I don't think this is needed? Can't we just delete this everywhere and do `socketServer.io.of('/workspace').to(workspace_id).emit(...)`?